src/main/java/net/minecraftforge/client/model/ModelLoader.java

```diff
@@ -336,7 +417,15 @@ public WeightedRandomModel(Variants variants)
             {
                 ResourceLocation loc = v.getModelLocation();
                 locations.add(loc);
-                IModel model = new WeightedPartWrapper(getModel(loc));
+
+                IModel model = getModel(loc);
+                if (v instanceof ISmartVariant)
+                {
+                    model = ((ISmartVariant)v).process(model, ModelLoader.this);
```

Comment:
I don't see any way of referencing these variant-applied models in code/in json, yet I think it should be useful. Maybe a way of constructing and resolving resource locations referring to the blockstate definitions directly (and adding a possibility to recursively apply blockstate modifications) is a good idea.